src/stream/src/executor/top_n/topn_cache_state.rs

```diff
@@ -108,18 +108,25 @@ impl TopNCacheState {
         self.inner.range(range)
     }
 
-    pub fn extract_if<F>(&mut self, pred: F) -> ExtractIf<'_, CacheKey, CompactedRow, F, Global>
+    pub fn extract_if<'a, F1>(
```

Comment:
This change makes me think:
1. Are there any other places that the size is not correctly maintained?
2. Why this is `kv_heap_size` stuff is in `TopNCacheState`? It looks like a general `BTreeMapWithEstimateSize` or sth.

Reply:
There are some special usage of an estimated structure. For example https://github.com/singularity-data/risingwave/blob/9cd7f6486e659af72857aef7ac27247305e880c0/src/stream/src/executor/managed_state/join/join_entry_state.rs#L75. It can be hard to provide an EstimatedBtreeMap with an unified semantic.